build.zig

```diff
@@ -108,36 +112,47 @@ pub fn build(b: *std.Build) !void {
     lib.installHeader(pcre2_header, "pcre2.h");
     b.installArtifact(lib);
 
-
     // pcre2test
 
-    const pcre2test = b.addExecutable(.{
-        .name = "pcre2test",
+    const pcre2test_mod = std.Build.Module.create(b, .{
         .target = target,
         .optimize = optimize,
+        .link_libc = true,
+        .imports = &.{std.Build.Module.Import{ .name = b.fmt("pcre2-{s}", .{@tagName(codeUnitWidth)}), .module = lib_mod }},
     });
 
+    pcre2test_mod.addCMacro("HAVE_CONFIG_H", "");
+
+    const pcre2test = b.addExecutable(.{
+        .name = "pcre2test",
+        .root_module = pcre2test_mod,
+        .linkage = linkage,
```

Comment:
What does "linkage" mean for an executable? Altering the linkage to choose between `libpcre2-8.{so,a}` shouldn't force a property change on the pcre2test binary.

Reply:
another great question. This was likely added in an effort to fix some error messages when I  hadn't yet removed the previous `linkLibrary()` steps.

I inferred the necessity by the executeable simply having the field at all, and the fact that we were 'linking libraries'. But once I switched to Importing the libraries into the pcre2test module, that likely was no longer a meaningful inference. I'll also test this now.

Reply:
yes it does seem like this is a redundant line, likely because we are importing now, instead of linking. likely whatever error it initially suppressed was a red herring.